master/internal/authz/permission_denied_error.go

```diff
@@ -0,0 +1,28 @@
+package authz
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/determined-ai/determined/proto/pkg/rbacv1"
+)
+
+// PermissionDeniedError represents an error that arises when a user does not have sufficient
+// access privileges. RequiredPermissions can be empty for non-rbac errors.
+type PermissionDeniedError struct {
+	RequiredPermissions []*rbacv1.Permission
+}
+
+// Error returns an error string.
+func (p PermissionDeniedError) Error() string {
+	if len(p.RequiredPermissions) == 0 {
+		return "access denied"
+	}
+
+	permissions := make([]string, len(p.RequiredPermissions))
+	for i, perm := range p.RequiredPermissions {
+		permissions[i] = perm.Name
+	}
+	return fmt.Sprintf("access denied; required permissions: %s", strings.Join(
```

Comment:
should we say what scope you need it on / or globally?

In the case of ```moveProject``` you need a delete on one workspace and a create on another so just saying ```required permissions: DELETE, CREATE``` isn't as specific as we could be

Reply:
I had a chat with Max about this yesterday and the consensus was that you know what operation you're performing and on what scope(s), so you would need to know what permissions you require to carry out that operation.